### ISSUE TITLE
Stuttering on Arch Linux

### ISSUE THREAD

**Issue by skippa** [love2d/love#1630]
I have a small demo 2d platformer - pretty basic - uses anim8 for sprites and tiles.  Works fine on one Arch Linux system, but momentarily pauses every 5 seconds or so, on another Arch Linux system.  Same hardware...

What logs / debugging can I do to find out whats causing the issue?  What video system does Love2D use for its linux binaries?

Thanks!

**Comment by MikuAuahDark** [comment]
I tested very simple program that ping pongs a circle horizontally and I can't reproduce the freezing under my Arch Linux.

LOVE uses SDL for windowing and OpenGL for graphics. The driver used however depends on your system.

**Comment by skippa** [comment]
Thanks, I'm looking into my OpenGL drivers - it seems to be something weird with one of my Arch Linux installations - as when I boot from my USB Arch Linux with same hardware - the issue goes away.  It could also be something specific to the code I'm using - the only external libraries I use are push.lua, class.lua and anim8.  When I run Love without a game, there is not glitching on the 'demo' animation...

**Comment by MikuAuahDark** [comment]
You can try to compile LOVE with profiler flags on and see which function takes the most time.

**Comment by scurest** [comment]
I have the same issue with love-11.3-x86_64.AppImage from the 11.3 release. It doesn't happen with (1) love-11.2-x86_64.AppImage from the 11.2 release, or (2) love compiled from source, either from master or from the 11.3 tag.

**Comment by skippa** [comment]
@scurest yes, that is exactly my problem too!  I just downloaded the 11.2 AppImage and sure enough it doesn't happen.  Glad to hear it's not just me ;)  You running Arch Linux too, right?

@MikuAuahDark is that the only way to get love to debug to the console?

I've spent the last few hours going through all my Arch packages trying to work out which is the offending package (as I said earlier, one Arch Linux install works fine, the other does not - exact same hardware - but different settings/packages/etc).  Video drivers, VA-API and OpenGL drivers are all identical.  This is a strange issue for sure

**Comment by MikuAuahDark** [comment]
No, that's one way to measure which C function takes the most time.

Out of curiosity, does this happends when you're compiling the latest `master` branch?

**Comment by scurest** [comment]
@skippa Yeah, on Arch.

This seems to be caused by the SDL distributed with 11.3.

````sh
wget https://github.com/love2d/love/releases/download/11.3/love-11.3-linux-x86_64.tar.gz
aunpack love-11.3-linux-x86_64.tar.gz
cd love-11.3-linux-x86_64
LD_LIBRARY_PATH=dest/usr/lib dest/usr/bin/love <game>  # stutters
````

It also stutters if I use a love compiled from master as long as I use the same libraries

````sh
LD_LIBRARY_PATH=dest/usr/lib <love compiled from master> <game>  # also stutters
````

We can narrow it down to SDL by replacing with my system's copy (2.0.12-2)

````
cp /usr/lib/libSDL2-2.0.so.0 dest/usr/lib/libSDL2-2.0.so.0
LD_LIBRARY_PATH=dest/usr/lib dest/usr/bin/love ~/te  # no stuttering
````

**Comment by scurest** [comment]
Possibly related to https://bugzilla.libsdl.org/show_bug.cgi?id=4417? 3000ms matches up and adding `t.modules.joystick = false` to conf.lua makes the stutter go away.

**Comment by MikuAuahDark** [comment]
> If love 12 has moved from SDL 2.0.10 to 2.0.12 this might be fixed already (I don't know how to check that).

This depends entirely on the Linux distro that provides SDL. If the distros you're using didn't use SDL 2.0.12 then LOVE just use whatever SDL version provided by the system (assume compiled from source or provided by your distro). However I don't think this is an issue for ArchLinux.

For the AppImage itself, LOVE 11.3 was released before SDL 2.0.12 is released, so the AppImage uses SDL < 2.0.12. I'm not too familiar with the AppImage packaging of the GitHub actions artifacts but maybe that's worth to try.

Since this is problem related to SDL, I'll mark this as closed.

**Comment by scurest** [comment]
> This depends entirely on the Linux distro that provides SDL.

Right, this is a problem for the AppImage though.

> GitHub actions artifacts

I downloaded an AppImage from the first PR, namely this: https://github.com/love2d/love/suites/1207720396/artifacts/17931421. This contains a libSDL that's version 2.0.4 (the bug I linked says the issue was not present before 2.0.9). So that means the process for building the AppImage for Github actions isn't the same as the process for building the AppImage for releases. It would be best if these were identical; otherwise, you can test the artifacts, but it won't guarantee there's no problem with the release.

> Since this is problem related to SDL which LOVE can't fix, I'll mark this as closed.

But LOVE should make sure that when 12 is released, the AppImage will include SDL >=2.0.12.
